src/services/document-drive/electron-document-drive.ts

```diff
@@ -0,0 +1,231 @@
+            this.serialize()
+        );
+    }
+
+    serialize() {
+        return JSON.stringify({
+            basePath: this.basePath,
+            document: this.document.toDocument() as Document<
+                DocumentDriveState,
+                DocumentDriveAction
+            >,
+        } satisfies IElectronDocumentDriveSerialized);
+    }
+
+    static deserialize(input: string, store: ElectronStore) {
+        try {
+            const { basePath, document } = JSON.parse(
+                input
+            ) as IElectronDocumentDriveSerialized;
+            return new ElectronDocumentDrive(
+                store,
+                basePath,
+                document // TODO
+            );
+        } catch (error) {
+            return undefined;
+        }
+    }
+
+    static load(store: ElectronStore) {
+        const result = store.get(
+            ElectronDocumentDrive.DOCUMENT_DRIVE_KEY
+        ) as string;
+        if (!result) {
+            return undefined;
+        }
+        return this.deserialize(result, store);
+    }
+
+    private async saveFile(driveId: string, path: string, document: Document) {
+        return fs.writeFile(
+            this.getPath(driveId, path),
+            JSON.stringify(document),
+            {
+                encoding: 'binary',
+            }
+        );
+    }
+
+    private async deleteFile(driveId: string, path: string) {
+        return fs.rm(this.getPath(driveId, path), { recursive: true });
+    }
+
+    async addFolder(input: AddFolderInput) {
+        await fs.mkdir(this.getPath(input.drive, input.path, ''));
+        this.document.addFolder(input);
+        this.save();
+
+        const node = this.getNode(input.drive, input.path);
+        if (!node || !isFolderNode(node)) {
+            throw new Error('Error adding folder');
+        }
+
+        return node;
+    }
+
+    async addFile(input: AddFileInput, document: Document) {
+        await this.saveFile(input.drive, input.path, document);
+        this.document.addFile(input);
+
+        const node = this.getNode(input.drive, input.path);
+        if (!node || !isFileNode(node)) {
+            throw new Error('Error adding file');
+        }
+
+        this.save();
+        return node;
+    }
+
+    async updateFile(input: UpdateFileInput, document: Document) {
+        await this.saveFile(input.drive, input.path, document);
+        this.document.updateFile(input);
+    }
+
+    async deleteNode(drive: string, path: string) {
+        await this.deleteFile(drive, path);
+        this.document.deleteNode({ drive, path });
+        this.save();
+    }
+
+    async renameNode(drive: string, path: string, name: string) {
+        const node = this.getNode(drive, path);
+        if (!node) {
+            throw new Error(
+                `Node with path ${path} not found on drive ${drive}`
+            );
+        }
+
+        this.document.updateNode({ drive, path, name });
```

Comment:
We just change the name of the node but keep the same path as it is the Id of the node. The `updateNode` operation uses the path to know which node to update so we would need to add a `newPath` parameter to the operation to support also changing the path. We could also create a new "RENAME_NODE" operation that would support this.

Reply:
what if we use a different id instead the path?

Reply:
We had a separate Id field before but changed to use path. I guess this would be an use for the Id property.
On the other hand it fills a bit weird to use an id instead of a path in some of the operations.